### DISCUSSION TITLE
server side pagination is not updated

### DISCUSSION THREAD

**Discussion by aniket-solulab** [vercel/next.js#58326] [App Router]
i have one blog page with post lost and a pagination component my blog page is a server component.
the issue i am facing is on changing the page the the query should be refreshed and hence the server rendered page should also be refreshed but here i am getting the previous array data and new array data from the server side components 

https://github.com/vercel/next.js/assets/85621201/76024d19-d778-4977-a33c-d5a65edcd93e

please help me i am new 

here is blogs page.tsx

export const metadata = {
  title: 'Blog',
}
export const revalidate = 0;

export default async function BlogPage({ searchParams }) {
  const searchkey = searchParams['search'];
  const page = searchParams['page'] ?? '1';

  const allposts: { blogs: bloginterface[], total: number, totalPages: number, currentpage: number } = await getAllposts(page, searchkey)
  return (
    <>
      <Blogheader />
      {
        searchkey && (
          <p className='my-3 '>Showing results for keyword &ldquo;{searchkey}&ldquo;</p>
        )
      }
      {allposts?.blogs?.length ? (
        <div className='blogpostgrid mt-2 cp'>
          {allposts?.blogs?.map((post: bloginterface, index) => (
            <>
              <div
                key={post._id}
                className='group relative flex flex-col space-y-2 desktopwrapper'
              >
                {post.previewImage && (
                  <Image
                    src={post.previewImage}
                    unoptimized
                    alt={post.title}
                    width={100}
                    height={100}
                    className='rounded-sm border postimage bg-muted transition-colors'
                    priority={index <= 1}
                  />
                )}
                <h2 className='text-2xl posttitle my-2'>{concatString(61, post.title)}</h2>
                {post.description && (
                  <p className='text_muted my-1'>{concatString(86, post.description)}</p>
                )}
                {post.createdAt && (
                  <div className='wrapper justify-between'>
                    <p className='text-sm my-1 text-muted-foreground'>
                      {formatDate(post.createdAt)}
                    </p>
                    <Link prefetch={false} href={`blog/posts/${post.slug}`} className='absolute viewarticle inset-0 cp'>
                      <span className=''>View Article</span>
                    </Link>
                  </div>
                )}
              </div>
              <Link
                key={post._id}
                prefetch={false} href={`blog/posts/${post.slug}`}
                className='relative wrapper mobliewrapper'
              >
                <div >
                  <div className='wrapper justify-start '>
                    <Image src={post.author[0].profilePic} alt={post.author[0].name} className='rounded-pill' width={28} height={28} unoptimized />
                    <p className='mb-0 ms-2 usernameblog' > {post.author[0].userName} </p>
                    <p className='ms-2 text_muted mb-0 datepostmob' > {moment(post.createdAt).format("MMM Do YY")} </p>
                  </div>
                  <h2 className='text-2xl posttitle 
                  '>{concatString(70, post.title)}</h2>
                  <p className='text_muted post-subtitle my-1'>{concatString(110, post.description)}</p>
                  {/* <div className='wrapper justify-between'>
                    <Link prefetch={false} href={`blog/posts/${post.slug}`} className='absolute viewarticle inset-0 cp'>
                      <span className=''>View Article</span>
                    </Link>
                  </div> */}
                </div>
                {post.previewImage && (
                  <div className='mobileimgwrapper ms-2'>
                    <Image
                      src={post.previewImage}
                      unoptimized
                      alt={post.title}
                      width={80}
                      height={56}
                      className='rounded-sm border postimagemobile bg-muted transition-colors'
                      priority={index <= 1}
                    />
                  </div>
                )}
              </Link>
            </>
          ))}
        </div>
      ) : (
        <p>No posts published.</p>
      )}
      <BlogPagination currentpage={allposts.currentpage} total={allposts.total} totalPages={allposts.totalPages} />
    </>
  
)
}

blogpagination 

const BlogPagination: React.FC<{ total: number, totalPages: number, currentpage: number, }> = ({ total, totalPages, currentpage }) => {

    const { PaginationCOnfig, setpaginationConfig, currentPage, setCurrentPage } = usePaginationHook();
    const router = useRouter()
    const params = useSearchParams()
    useEffect(() => {
        if (currentPage) {
            router.push(`/blog?page=${currentPage}${params.get("search") ? `&search=${params.get("search")}` : ""}`)
        }
    }, [currentPage,])

    useEffect(() => {
        setpaginationConfig({
            itemCount: total,
            pagecount: totalPages,
            currentpage: currentpage,
            perpageitemcount: 10,
        });
        setCurrentPage((currentpage));
    }, [totalPages, total,])
    return (
        <div className='wrapper w-100'>
            {PaginationCOnfig && (
                <Col className="col-12">
                    <Pagination showRowperpage={false} paginationConfig={PaginationCOnfig as PaginationConfigType} />
                </Col>
            )}
        </div>
    )
}

export default BlogPagination

**Comment by samohovets** [comment]
Same thing, `v14.0.3-canary.7` or `14.0.2`. Had to change my links to <a> to push the temporary fix. Looks like the script responsible for client-side navigation not doing its job.

**Reply by samohovets** [reply]
I was able to fix it by setting `prefetch={false}` on my links. The issue happened after I upgraded from 13.5 to 14.0.2.

**Reply by aniket-solulab** [reply]
Ok but i have used router. push for changing the link should I replace it with link . Can see in my blogpagination code above

**Reply by BenjaOliva** [reply]
Hi, facing same issue on a paginated list and changing page seaechParam. So just setting the prefetch false did the trick or I have to use Link on my buttons for pagination to work as expected.

Really annoyed cause it works flawlessly in local but not at hosted site
